### PR TITLE
Added Download Button

### PR DESCRIPTION
**Pull request by dimi-23**
**Fixes issue:**

#59 

**Changes:**

A download button has been added to download the code easily.

image

### CODE REVIEW

**Comment by Sakshi183** [comment]
This is an offline chrome extension .So, all the algorithms are already available offline.So, I guess the option to download code is not required. Instead, we can open the directory where the code is stored for editing the code.

**Comment by dimi-23** [comment]
Hi @Sakshi183 

Yes, Quark is offline but I added the feature keeping in mind the following points:

1. Codes available are accessible only from the Chrome browser. If the users are given an option to save it to their desired directory, it will help them when the browser is not accessible. 

2. Accessing extension's directory is a tedious job and no user will do that. The users can copy-paste the code if they need it to run on their local compilers, but if a download option is present it can make the process faster.

3. If the download option is not directly available and the users try to access the codes from their default directory, they may mishandle and misplace some codes, resulting in undesirable behavior of the extension.

So, keeping in mind these points this issue was raised and implemented.

@AdiChat please correct me if I am wrong in my assessment.

**Comment by AdiChat** [comment]
Yes, a download feature is useful. Users do have a tendency to download code and work on it locally. :+1:

File: code/js/popup.js [hunk 1]
```diff
@@ -154,9 +154,9 @@ function dumpBookmarks(query)
 					    {	  	
 	
 						   	if(!favs.includes(fname)) {
-						   		inside_text = inside_text + "<a  target='_blank' href='/code/"+key+"/"+obj[key][dd]+"'>"+sub_result_number+". "+obj[key][dd]+"</a>"+"&nbsp;&nbsp;<i id='myStar"+temp2+"\' class='fa fa-star'></i><br>";
+						   		inside_text = inside_text + "<a  target='_blank' href='/code/"+key+"/"+obj[key][dd]+"'>"+sub_result_number+". "+obj[key][dd]+"</a>"+"&nbsp;&nbsp;<a style='color:inherit' href='/code/"+key+"/"+obj[key][dd]+"'download><i id='myDownload"+temp2+"\' style='float:right' class='fa fa-download'></i></a><i id='myStar"+temp+"\' style='float:right;width:8%' class='fa fa-star'></i><br>";
```
Comment (AdiChat) [inline code/js/popup.js]: It looks like `temp` should be replaced by `temp2`. This is breaking the favorites features. Same is applicable to the else section. :+1: 

Kindly take a look into this. :+1:

Comment (dimi-23) [reply code/js/popup.js]: Oops yes! Made the corresponding changes.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**